qutebrowser/mainwindow/tabbedbrowser.py

```diff
@@ -584,10 +584,14 @@ def on_load_started(self, tab):
     @pyqtSlot()
     def on_cur_load_started(self):
         """Leave insert/hint mode when loading started."""
-        modeman.leave(self._win_id, usertypes.KeyMode.insert, 'load started',
-                      maybe=True)
-        modeman.leave(self._win_id, usertypes.KeyMode.hint, 'load started',
-                      maybe=True)
+        if config.instance.get('input.insert_mode.leave_on_load',
+                               url=self.current_url()):
+            modeman.leave(self._win_id, usertypes.KeyMode.insert,
+                          'load started', maybe=True)
+            modeman.leave(self._win_id, usertypes.KeyMode.hint,
+                          'load started', maybe=True)
```

Comment:
The hint mode should probably be left either way?

Reply:
(just for the record, I would prefer the hint mode to not be left (as the qt bug clears hints as well currently, but I am ok with putting that behind another knob or similar)).

Reply:
@jgkamat Do you have a real-life scenario where the Qt bug is an issue with hint mode? From what I've seen so far, it only happens when e.g. typing, which you usually don't do when in hint mode.

Reply:
I have some sites where the qt bug fires almost constantly, so I can't do any hinting at all (and the disk gets thrashed as well currently).

More common example: Use the mouse to drag openstreetmap, and then switch to the kb to hint search. If you aren't fast enough, you'll loose hinting.